app/controllers/leaderboards_controller.rb

```diff
@@ -2,105 +2,39 @@ class LeaderboardsController < ApplicationController
-                                       .where("leaderboards.updated_at >= ?", 30.days.ago)
-                                       .where.not(user_id: monthly_top_three_user_ids)
-                                       .order(points: :desc)
+  def top_three_users
+    time_range = determine_time_range(params[:range])
+    @top_three_users = Leaderboard.top_three_users(time_range)
 
-    if @monthly_leaderboards.any?
-      render json: @monthly_leaderboards
+    if @top_three_users.any?
+      render json: @top_three_users
     else
-      render json: { message: 'No monthly leaderboard available' }, status: :ok
+      render json: { message: "No top three users available for the selected range" }, status: :ok
     end
   end
 
-  def weekly_top_three_users
-    @weekly_top_three_users = Leaderboard.joins(:user)
-                                          .select("leaderboards.id, users.id AS user_id, users.full_name, leaderboards.points, leaderboards.created_at, leaderboards.updated_at")
-                                          .where("leaderboards.updated_at >= ?", 7.days.ago)
-                                          .order(points: :desc)
-                                          .limit(3)
+  def leaderboards
+    time_range = determine_time_range(params[:range])
+    @leaderboards = Leaderboard.leaderboards_excluding_top_three(time_range)
 
-    if @weekly_top_three_users.any?
-      render json: @weekly_top_three_users
+    if @leaderboards.any?
+      render json: @leaderboards
     else
-      render json: { message: 'No weekly top three users available' }, status: :ok
+      render json: { message: "No leaderboard available for the selected range" }, status: :ok
     end
   end
 
-  def weekly_leaderboards
-    weekly_top_three_user_ids = Leaderboard.joins(:user)
-                                           .where("leaderboards.updated_at >= ?", 7.days.ago)
-                                           .order(points: :desc)
-                                           .limit(3)
-                                           .pluck("users.id")
-
-    @weekly_leaderboards = Leaderboard.joins(:user)
-                                      .select("leaderboards.id, users.id AS user_id, users.full_name, leaderboards.points, leaderboards.created_at, leaderboards.updated_at")
-                                      .where("leaderboards.updated_at >= ?", 7.days.ago)
-                                      .where.not(user_id: weekly_top_three_user_ids)
-                                      .order(points: :desc)
-
-    if @weekly_leaderboards.any?
-      render json: @weekly_leaderboards
+  private
+  def determine_time_range(range)
+    case range
+    when "daily"
+      1.day.ago
+    when "weekly"
+      7.days.ago
+    when "monthly"
+      30.days.ago
     else
-      render json: { message: 'No weekly leaderboard available' }, status: :ok
+      30.days.ago
     end
   end
```

Comment:
I think its better if you declare this method to model level.

Reply:
Moved this to leaderboard model.